File: CSharp/Tests/GetFileExtensionTest.cs
using Xunit;

namespace CSharp.Tests
{
    public class GetFileExtensionTest
    {
        [Theory]
        [InlineData(new string[] { "project1.jpg", "project1.pdf", "project1.mp3" }, new string[] { "jpg", "pdf", "mp3" })]
        [InlineData(new string[] { "ruby.rb", "cplusplus.cpp", "python.py", "javascript.js" }, new string[] { "rb", "cpp", "py", "js" })]
        [InlineData(new string[] { "code.html", "code.css" }, new string[] { "html", "css" })]
        [InlineData(new string[] { "minesweeper.java", "game.java", "my_project.java" }, new string[] { "java", "java", "java" })]
        public void GetExtension_FileNameStringArrayInput_ReturnSringArrayOfExtensionNames(string[] arr, string[] expected)
        {
            var actual = GetFileExtension.GetExtension(arr);

            Assert.Equal(expected, actual);
        }
    }
}

File: js/tests/getFileExtension.test.js
const getExtension = require("../getFileExtension");

const data = [
  [
    ["project1.jpg", "project1.pdf", "project1.mp3"],
    ["jpg", "pdf", "mp3"]
  ],
  [
    ["ruby.rb", "cplusplus.cpp", "python.py", "javascript.js"],
    ["rb", "cpp", "py", "js"]
  ],
  [
    ["code.html", "code.css"],
    ["html", "css"]
  ],
  [
    ["minesweeper.java", "game.java", "my_project.java"],
    ["java", "java", "java"]
  ]
];

describe("getExtension", () => {
  describe("Return array of file extensions", () => {
    test.each(data)("getExtension(%p) = %p", (arr, result) => {
      let actual = getExtension(arr);

      expect(actual).toEqual(result);
    });
  });
});
